src/test/scala/ai/privado/languageEngine/csharp/CSharpTestBase.scala

```diff
@@ -0,0 +1,195 @@
+
+  private val cpgs        = mutable.ArrayBuffer.empty[Cpg]
+  private val outPutFiles = mutable.ArrayBuffer.empty[File]
+  private val inputDirs   = mutable.ArrayBuffer.empty[File]
+
+  val sourceRule: List[RuleInfo] = List(
+    RuleInfo(
+      "Data.Sensitive.PersonalIdentification.FirstName",
+      "FirstName",
+      "",
+      FilterProperty.METHOD_FULL_NAME,
+      Array(),
+      List("(?i).*firstName.*"),
+      false,
+      "",
+      Map(),
+      NodeType.REGULAR,
+      "",
+      CatLevelOne.SOURCES,
+      "",
+      Language.UNKNOWN,
+      Array()
+    ),
+    RuleInfo(
+      "Data.Sensitive.PersonalIdentification.LastName",
+      "LastName",
+      "",
+      FilterProperty.METHOD_FULL_NAME,
+      Array(),
+      List("(?i).*lastName.*"),
+      false,
+      "",
+      Map(),
+      NodeType.REGULAR,
+      "",
+      CatLevelOne.SOURCES,
+      "",
+      Language.UNKNOWN,
+      Array()
+    ),
+    RuleInfo(
+      "Data.Sensitive.PersonalIdentification.DateofBirth",
+      "Date of Birth",
+      "",
+      FilterProperty.METHOD_FULL_NAME,
+      Array(),
+      List("(?i).*dob.*"),
+      false,
+      "",
+      Map(),
+      NodeType.REGULAR,
+      "",
+      CatLevelOne.SOURCES,
+      "",
+      Language.UNKNOWN,
+      Array()
+    ),
+    RuleInfo(
+      "Data.Sensitive.ContactData.EmailAddress",
+      "EmailAddress",
+      "",
+      FilterProperty.METHOD_FULL_NAME,
+      Array(),
+      List("(?i).*email.*"),
+      true,
+      "",
+      Map(),
+      NodeType.REGULAR,
+      "",
+      CatLevelOne.SOURCES,
+      "",
+      Language.UNKNOWN,
+      Array()
+    ),
+    RuleInfo(
+      "Data.Sensitive.ContactData.PhoneNumber",
+      "Phone",
+      "",
+      FilterProperty.METHOD_FULL_NAME,
+      Array(),
+      List("(?i).*phone.*"),
+      true,
+      "",
+      Map(),
+      NodeType.REGULAR,
+      "",
+      CatLevelOne.SOURCES,
+      "",
+      Language.UNKNOWN,
+      Array()
+    ),
+    RuleInfo(
+      "Data.Sensitive.FinancialData.Salary",
+      "Salary",
+      "",
+      FilterProperty.METHOD_FULL_NAME,
+      Array(),
+      List("(?i).*salary.*"),
+      true,
+      "",
+      Map(),
+      NodeType.REGULAR,
+      "",
+      CatLevelOne.SOURCES,
+      "",
+      Language.UNKNOWN,
+      Array()
+    )
+  )
+
+  val sinkRules: List[RuleInfo] = List(
+    RuleInfo(
+      "Loggers.Console",
+      "WriteLine",
+      "",
+      FilterProperty.METHOD_FULL_NAME,
+      Array(),
+      List("(?i).*WriteLine.*"),
+      false,
+      "",
+      Map(),
+      NodeType.REGULAR,
+      "",
+      CatLevelOne.SINKS,
+      "",
+      Language.CSHARP,
+      Array()
+    )
+  )
+
+  val configAndRules: ConfigAndRules =
+    ConfigAndRules(sourceRule, sinkRules, List(), List(), List(), List(), List(), List(), List(), List())
+
+  val taggerCache = new TaggerCache()
+
+  def code(code: String, fileExtension: String = ".cs"): (Cpg, ThreatEngineExecutor) = {
```

Comment:
This is good, but I would recommend something like this to accommodate multiple files as well
- https://github.com/Privado-Inc/privado-core/blob/main/src/test/scala/ai/privado/languageEngine/ruby/RubyTestBase.scala